services/secondary-storage/server/src/org/apache/cloudstack/storage/resource/NfsSecondaryStorageResource.java

```diff
@@ -525,6 +525,8 @@ protected Answer copySnapshotToTemplateFromNfsToNfs(CopyCommand cmd, SnapshotObj
                     bufferWriter.write("\n");
                     long size = _storage.getSize(destFileFullPath);
                     bufferWriter.write("size=" + size);
+                    bufferWriter.close();
```

Comment:
@niteshsarda I see what you're saying, however I disagree. You can refactor the code in either multiple try-with-resource blocks or methods in such blocks, save the `template.properties` file first and use the processor/template-location in another block. FWIW, it's still acceptable.